vignettes/package.Rmd

```diff
@@ -2,119 +2,250 @@
 title: "Using reticulate in an R Package"
 output:
   rmarkdown::html_vignette
+editor_options:
+  markdown:
+    wrap: 80
 vignette: >
   %\VignetteIndexEntry{Using reticulate in an R Package}
   %\VignetteEngine{knitr::rmarkdown}
   %\VignetteEncoding{UTF-8}
 ---
 
+## Declaring Python Requirements
 
-## Delay Loading Python Modules
+Beginning with Reticulate version 1.41, R packages can declare Python
+requirements with `py_require()`. Reticulate can then resolve these requirements
+at runtime using an ephemeral Python environment.
```

Comment:
What exactly is `py_require()` doing? When do the requested Python packages get loaded?

Reply:
`py_require()` calls build up a manifest of requirements internally. The manifest is used when reticulate is initializing Python. The manifest is onl used to resolve an ephemeral virtual environment if no self-managed python installation is higher in the order of discovery.